`GetDocFiles` may run more than once

`GetDocFiles` may run more than once:
https://github.com/dotnet/winforms/blob/725161305d1c0fe9a88914ae81466dc79dad3f38/eng/packageContent.targets#L23

This may cause build failures due to a race condition, e.g., [this build](https://dev.azure.com/dnceng/public/_build/results?buildId=1679927&view=results):
```
dotnet-install: Installation finished
  Determining projects to restore...
  Determining projects to restore...
  Determining projects to restore...
  Restored D:\a\_work\1\s\src\System.Windows.Forms\tests\UnitTests\WindowsFormsTests\WindowsFormsTests.proj (in 7.91 sec).
  Restored D:\a\_work\1\s\src\System.Windows.Forms\tests\InteropTests\NativeTests\NativeTests.proj (in 7.65 sec).
D:\a\_work\1\s\eng\packageContent.targets(55,5): error MSB3026: Could not copy "D:\a\_work\1\s\.packages\microsoft.private.intellisense\7.0.0-preview-20220316.2\IntellisenseFiles\windowsdesktop\1033\Accessibility.xml" to "D:\a\_work\1\s\artifacts\bin\docs\Accessibility.xml". Beginning retry 1 in 1000ms. The process cannot access the file 'D:\a\_work\1\s\artifacts\bin\docs\Accessibility.xml' because it is being used by another process.  [D:\a\_work\1\s\src\System.Windows.Forms\tests\InteropTests\NativeTests\NativeTests.proj]
##[error]eng\packageContent.targets(55,5): error MSB3026: (NETCORE_ENGINEERING_TELEMETRY=Restore) Could not copy "D:\a\_work\1\s\.packages\microsoft.private.intellisense\7.0.0-preview-20220316.2\IntellisenseFiles\windowsdesktop\1033\Accessibility.xml" to "D:\a\_work\1\s\artifacts\bin\docs\Accessibility.xml". Beginning retry 1 in 1000ms. The process cannot access the file 'D:\a\_work\1\s\artifacts\bin\docs\Accessibility.xml' because it is being used by another process. 
D:\a\_work\1\s\eng\packageContent.targets(55,5): error MSB3026: Could not copy "D:\a\_work\1\s\.packages\microsoft.private.intellisense\7.0.0-preview-20220316.2\IntellisenseFiles\windowsdesktop\1033\Accessibility.xml" to "D:\a\_work\1\s\artifacts\bin\docs\Accessibility.xml". Beginning retry 1 in 1000ms. The process cannot access the file 'D:\a\_work\1\s\artifacts\bin\docs\Accessibility.xml' because it is being used by another process.  [D:\a\_work\1\s\src\System.Windows.Forms\tests\InteropTests\NativeTests\NativeTests.proj]
##[error]eng\packageContent.targets(55,5): error MSB3026: (NETCORE_ENGINEERING_TELEMETRY=Restore) Could not copy "D:\a\_work\1\s\.packages\microsoft.private.intellisense\7.0.0-preview-20220316.2\IntellisenseFiles\windowsdesktop\1033\Accessibility.xml" to "D:\a\_work\1\s\artifacts\bin\docs\Accessibility.xml". Beginning retry 1 in 1000ms. The process cannot access the file 'D:\a\_work\1\s\artifacts\bin\docs\Accessibility.xml' because it is being used by another process. 
```

[BuildSrc-x86.zip](https://github.com/dotnet/winforms/files/8370456/BuildSrc-x86.zip)

/cc: @singhashish-wpf for visibility

--- a/eng/Versions.props
+++ b/eng/Versions.props
@@ -48,7 +48,7 @@
   </PropertyGroup>
   <!-- Docs / Intellisense -->
   <PropertyGroup>
-    <MicrosoftPrivateIntellisenseVersion>7.0.0-preview-20220316.2</MicrosoftPrivateIntellisenseVersion>
+    <MicrosoftPrivateIntellisenseVersion>7.0.0-preview-20220331.1</MicrosoftPrivateIntellisenseVersion>
   </PropertyGroup>
   <!-- Arcade -->
   <PropertyGroup>

--- a/eng/packageContent.targets
+++ b/eng/packageContent.targets
@@ -3,61 +3,12 @@
   <PropertyGroup>
     <_DotnetApiDocsPackageName>Microsoft.Private.Intellisense</_DotnetApiDocsPackageName>
     <!-- Using ToLower below to unlock Linux build, even if it's unsupported. -->
-    <_DotnetApiDocsFilesRoot>$([MSBuild]::NormalizeDirectory('$(NuGetPackageRoot)', '$(_DotnetApiDocsPackageName.ToLower())', '$(MicrosoftPrivateIntellisenseVersion)', 'IntellisenseFiles', 'windowsdesktop'))</_DotnetApiDocsFilesRoot>
-    <_IntellisenseXmlDir>$([MSBuild]::NormalizeDirectory('$(ArtifactsBinDir)', 'docs'))</_IntellisenseXmlDir>
+    <_DotnetApiDocsFilesRoot>$([MSBuild]::NormalizeDirectory('$(NuGetPackageRoot)', '$(_DotnetApiDocsPackageName.ToLower())', '$(MicrosoftPrivateIntellisenseVersion)', 'IntellisenseFiles', 'windowsdesktop', '1033'))</_DotnetApiDocsFilesRoot>
   </PropertyGroup>
 
   <ItemGroup>
     <PackageDownload Include="Microsoft.Private.Intellisense" Version="[$(MicrosoftPrivateIntellisenseVersion)]" />
   </ItemGroup>
-
- <!--
-    ============================================================
-                      GetDocFiles
-    Restore intellisense xml files from the centrally managed NuGet package that originates from https://github.com/dotnet/dotnet-api-docs.
-    This packages is published to 'dotnet6-transport' feed.
-
-    Note: The intellisense package doesn't use NuGet conventions so we need to select manually.
-    ============================================================
-    -->
-  <Target Name="GetDocFiles"
-          AfterTargets="Restore">
-
-    <ItemGroup>
-      <DocFile Include="$(_DotnetApiDocsFilesRoot)**\*.xml"/>
-      <DocFile>
-        <!-- trim off slash since it differs by platform and we need to do a string compare -->
-        <LCID>$([System.String]::new('%(RecursiveDir)').TrimEnd('\/'))</LCID>
-      </DocFile>
-      <DocFile>
-        <!-- map LCID to culture name that VS expects -->
-        <Culture>unknown</Culture>
-        <Culture Condition="'%(LCID)' == '1028'">zh-hant</Culture>
-        <Culture Condition="'%(LCID)' == '1031'">de</Culture>
-        <!-- english is placed in the root -->
-        <Culture Condition="'%(LCID)' == '1033'"></Culture>
-        <Culture Condition="'%(LCID)' == '1036'">fr</Culture>
-        <Culture Condition="'%(LCID)' == '1040'">it</Culture>
-        <Culture Condition="'%(LCID)' == '1041'">ja</Culture>
-        <Culture Condition="'%(LCID)' == '1042'">ko</Culture>
-        <Culture Condition="'%(LCID)' == '1049'">ru</Culture>
-        <Culture Condition="'%(LCID)' == '2052'">zh-hans</Culture>
-        <Culture Condition="'%(LCID)' == '3082'">es</Culture>
-      </DocFile>
-      <DocFile>
-        <SubFolder Condition="'%(Culture)' != ''">%(Culture)\</SubFolder>
-        <SubFolder Condition="'%(Culture)' == ''"></SubFolder>
-      </DocFile>
-    </ItemGroup>
-
-    <Error Condition="'%(DocFile.Culture)' == 'unknown'" Text="Unknown language folder '%(LCID)' for doc files '@(DocFile)'" />
-    
-    <Copy SourceFiles="@(DocFile)"
-          DestinationFiles="$(_IntellisenseXmlDir)%(SubFolder)%(FileName)%(Extension)"
-          SkipUnchangedFiles="true"
-          UseHardlinksIfPossible="true" />
-
-  </Target>
 
   <!--
     ============================================================
@@ -90,7 +41,8 @@
     </PropertyGroup>
 
     <PropertyGroup>
-      <IntellisenseXmlFileSource>$(_IntellisenseXmlDir)$(AssemblyName).xml</IntellisenseXmlFileSource>
+      <!-- The intellisense provided by the Docs system -->
+      <IntellisenseXmlFileSource>$([MSBuild]::NormalizePath('$(_DotnetApiDocsFilesRoot)', '$(AssemblyName).xml'))</IntellisenseXmlFileSource>
 
       <!-- Set the xml destination (for a later step that copies files from the $(_DotnetApiDocsPackageName) package to local build artifacts) -->
       <IntellisenseXmlDest Condition="'$(ProduceReferenceAssembly)' == 'true' And '$(PackageAsRefAndLib)' != 'true'" >$([System.IO.Path]::ChangeExtension('$(TargetRefPath)', '.xml'))</IntellisenseXmlDest>

